### PR TITLE
feat(api): Enforce apiLevel 2.15 for Flex-style deck slot names

### PR DESCRIPTION
**Pull request by SyntaxColoring**
# Overview

Closes RLAB-338.

# Test plan

Existing automated unit tests should be sufficient for this.

# Changelog

* Raise `APIVersionError` if you provide an OT-3–style deck slot name like `"A1"` to methods like `load_labware()`, and your protocol's `apiLevel` is less than `2.15`.
* Public-facing documentation (docs.opentrons.com) is *not* done in this PR. I'm doing that in RTC-286.

# Review requests

See comments.

# Risk assessment

Low.

### CODE REVIEW

File: api/src/opentrons/protocol_api/validation.py [hunk 1]
```diff
@@ -32,6 +34,9 @@
     from .labware import Well
 
 
+_COORDINATE_STYLE_DECK_LABELS_ADDED_IN = APIVersion(2, 15)
```
Comment (SyntaxColoring) [inline api/src/opentrons/protocol_api/validation.py]: Is there a better name for this? I feel like I'm inconsistent. "OT-3–style," "coordinate-style," "chess-style," ...

Comment (jbleon95) [reply api/src/opentrons/protocol_api/validation.py]: Battleship-style 🛥️ 

But seriously, I usually prefer including `VERSION_GATE` just to be more clear what this is being used for, so something like `_COORDINATE_DECK_LABEL_VERSION_GATE` would be good, imo

**Review by jbleon95 — APPROVED** [review]
LGTM, besides any name changes you want I think its good to go

**Review by jbleon95 — APPROVED** [review]
LGTM, besides any name changes you want I think its good to go

**Pull request merged**